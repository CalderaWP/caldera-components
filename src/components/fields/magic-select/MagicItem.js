import React from 'react';
import {
	optionShape
} from '../propTypes';
import classNames from 'classnames';
import PropTypes from 'prop-types';

/**
 * Create magic item for option of magic selects
 *
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const MagicItem = (props) => {
	return React.createElement(
		props.elementType,
		{
			style: {background: props.isHighlighted ? props.highlightColor : props.notHighlighterColor},
			className: classNames(props.className, 'magic-input-option')

		},
		(
			<React.Fragment>
				<span
					className={classNames('magic-item-label', 'magic-item-left')}

				>
				{props.item.value}
			</span>

				<span
					className={classNames('magic-item-value', 'magic-item-right')}
				>
				{props.item.label}
			</span>
			</React.Fragment>
		)
	);
};

/**
 * Prop definitions for MagicItem component
 *
 * @type {{item: shim, isHighlighted: shim, className: shim, highlightColor: shim, notHighlighterColor: shim}}
 */
MagicItem.propTypes = {
	elementType: PropTypes.oneOf(['div', 'span']),
	item: PropTypes.shape(optionShape),
	isHighlighted: PropTypes.bool,
	className: PropTypes.string,
	highlightColor: PropTypes.string,
	notHighlighterColor: PropTypes.string
};

/**
 * Default props for the MagicItem component
 *
 * @type {{isHighlighted: boolean, highlightColor: string, notHighlightedColor: string}}
 */
MagicItem.defaultProps = {
	elementType: 'div',
	isHighlighted: false,
	highlightColor: 'lightgray',
	notHighlightedColor: 'white'
};