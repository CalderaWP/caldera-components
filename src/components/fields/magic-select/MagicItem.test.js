import renderer from 'react-test-renderer';
import React from 'react';
import {MagicItem} from './MagicItem';


describe('MagicSelect component', () => {

	it('Matches snapshot', () => {
		const component = renderer.create(
			<MagicItem
				item={{
					label: 'HTML',
					value: 'html'
				}}
				highlightColor={'#fff00'}
				notHighlighterColor={'white'}
				isHighlighted={true}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Can render as span', () => {
		const component = renderer.create(
			<MagicItem
				item={{
					label: 'HTML',
					value: 'html'
				}}
				highlightColor={'#fff00'}
				notHighlighterColor={'white'}
				isHighlighted={true}
				elementType={'span'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});