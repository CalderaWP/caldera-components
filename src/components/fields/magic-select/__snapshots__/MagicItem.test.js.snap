// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicSelect component Can render as span 1`] = `
<span
  className="magic-input-option"
  style={
    Object {
      "background": "#fff00",
    }
  }
>
  <span
    className="magic-item-label magic-item-left"
  >
    html
  </span>
  <span
    className="magic-item-value magic-item-right"
  >
    HTML
  </span>
</span>
`;

exports[`MagicSelect component Matches snapshot 1`] = `
<div
  className="magic-input-option"
  style={
    Object {
      "background": "#fff00",
    }
  }
>
  <span
    className="magic-item-label magic-item-left"
  >
    html
  </span>
  <span
    className="magic-item-value magic-item-right"
  >
    HTML
  </span>
</div>
`;
